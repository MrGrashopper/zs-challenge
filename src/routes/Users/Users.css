.users {
    margin: 3rem 10%;
}

.users__row {
    display: flex;
    flex-direction: row;
    align-items: center;
}

.users__row li {
    list-style-type:none;
    padding: 0.5rem 1rem;
    width: 16rem;
}

.users__row button {
    background: #7dabb2;
    box-shadow: none;
    border-style: none;
    height: 1.5rem;
    width: 5rem;
    border-radius: 20px;
}

.users__row button:hover {
    background: red;
    cursor: pointer;
    color: white;
}