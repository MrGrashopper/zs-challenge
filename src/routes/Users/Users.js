import React, { useState, useEffect } from 'react';
import FetchUsers from "../../helpers/FetchUsers";
import { HandleDelete } from "../../helpers/Functions";
import './Users.css'; 

function Users() {
    const data = FetchUsers();
    const [users, setUsers] = useState();

    // delete user by click on button
    const handleDelete = (user) => {
        HandleDelete(user);

        if (users.includes(user)) {
            let list = [];
            users.filter(function(item) {
                if (item !== user) {
                    list.push(item)
                }
            })
            setUsers(list);
        }
    }

    useEffect(() => {
        setUsers(data.users)
    }, [data])

    useEffect(() => {
        users?.length >= 1 ? setUsers(users) : setUsers(null)
    }, [users])

    return (
        <div className="users">
            <div className="users__row">
            <ul className="users__row">
                        <li data-testid="email"><b>E-Mail</b></li>     
                        <li><b>First Name</b></li>           
                        <li><b>Last Name</b></li>  
                        <li><b>Action</b></li>  
                    </ul>            
            </div>
            { users && users.map((user, index) => {
                return(
                    <ul key={index} className="users__row">
                        <li>{ user.email }</li>     
                        <li>{ user.firstName }</li>           
                        <li>{ user.lastName }</li>  
                        <button onClick={() => handleDelete(user)}>delete</button>  
                    </ul>
                )
            })}
        </div>
    )
}

export default Users
