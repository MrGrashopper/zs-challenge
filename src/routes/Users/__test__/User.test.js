import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';

import Users from '../Users';

afterEach(cleanup);

describe('Test User component', () => {
  function tree() {
    return render(<Users/>)
  }

  it('should match with snapshot', () => {
    expect(tree()).toMatchSnapshot();
   });

   it('test button', () => {
     const { getByTestId } = tree();
     const element = screen.getByTestId('email');
     expect(element.textContent).toEqual("E-Mail")
     // screen.debug();
    });
  });