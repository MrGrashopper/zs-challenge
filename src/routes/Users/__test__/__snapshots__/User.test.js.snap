// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test User component should match with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="users"
      >
        <div
          class="users__row"
        >
          <ul
            class="users__row"
          >
            <li
              data-testid="email"
            >
              <b>
                E-Mail
              </b>
            </li>
            <li>
              <b>
                First Name
              </b>
            </li>
            <li>
              <b>
                Last Name
              </b>
            </li>
            <li>
              <b>
                Action
              </b>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="users"
    >
      <div
        class="users__row"
      >
        <ul
          class="users__row"
        >
          <li
            data-testid="email"
          >
            <b>
              E-Mail
            </b>
          </li>
          <li>
            <b>
              First Name
            </b>
          </li>
          <li>
            <b>
              Last Name
            </b>
          </li>
          <li>
            <b>
              Action
            </b>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
